Combine Dev Portal snippets and custom pages (#1668)
Kong/developer.konghq.com: app/dev-portal/pages-and-content.md

---
title: Dev Portal pages and content
content_type: reference
layout: reference

products:
    - dev-portal
beta: true
tags:
  - beta
works_on:
    - konnect
api_specs:
  - konnect/portal-management

search_aliases:
  - MDC
  - Markdown Components
  - Portal

breadcrumbs:
  - /dev-portal/

description: "Customize Dev Portal Pages using Markdown Components (MDC)."
faqs:
  - q: What visibility setting is applied when creating a new page?
    a: |
      Pages are created using the Default Visibility setting configured in your [Portal Settings](/dev-portal/portal-settings/).

  - q: Where can I find the URL for a page in Preview?
    a: |
      When previewing a page, the generated URL is shown at the bottom of the preview window.

  - q: What is the special behavior of the `home` page?
    a: |
      The `home` page represents the `/` root path of your Dev Portal. 
      If it is deleted, you’ll need to recreate it using the [Pages API](/api/konnect/portal-management/#/operations/create-portal-page).

>   - q: Is there a size limit for custom pages?
>     a: |
>       Yes. Custom pages are limited to a maximum of 1,000,000 characters.

Reviewer: ```suggestion
  - q: Is there a character limit for custom pages?
    a: |
      Yes. Custom pages are limited to a maximum of 1,000,000 characters.
```

  - q: Why aren’t parameterized values appearing in the page Preview?
    a: |
      The preview may not be able to display parameterized values. When the page is rendered in the Portal, parameters will be resolved. 
      Depending on the syntax used, the preview may not accurately reflect those values in Page or Snippet views.

  - q: Are snippets visible in the Portal by default?
    a: |
      Snippets are published by default, but they won’t appear in the Portal unless they are reused in a page.

  - q: What visibility setting is applied when creating a new snippet?
    a: |
      Snippets are created with the default page visibility setting configured in your [Portal settings](/dev-portal/portal-settings/).

>   - q: Is there a size limit for snippets?
>     a: |
>       Yes. Snippets are limited to a maximum of 1,000,000 characters.

Reviewer: ```suggestion
  - q: Is there a character limit for snippets?
    a: |
      Yes. Snippets are limited to a maximum of 1,000,000 characters.
```

Reviewer: Feels like 'character limit' is more precise here, but it's a minor detail.

related_resources:
  - text: Dev Portal Markdown components reference
    url: https://portaldocs.konghq.com/
  - text: Portal customization reference
    url: /dev-portal/portal-customization/
  - text: Custom domains
    url: /dev-portal/custom-domains/
---

In your Dev Portal, you can create pages that contain content such as text, buttons, tabs, and more. Pages are used to convey information about your API and Dev Portal to users. They are highly customizable using Markdown Components (MDC), allowing you to create nested page structures to organize pages and generate URLs or slugs. You can also stage new pages or restrict access to logged-in developers by using visibility controls and publishing status.

## Pages

> Like a website, you can create different pages for your Dev Portal. Pages can contain text and other objects, like containers and buttons. To get started creating pages, navigate to your Dev Portal and click **Portal Editor** in the sidebar. Pages are built using Markdown Components (MDC). Additional documentation on syntax, as well as tools for generating components, are available on a [dedicated MDC site](https://portaldocs.konghq.com/).

Reviewer: ```suggestion
You can create multiple pages in the Dev Portal, similar to how a website is structured. Pages can contain text and other objects, like containers and buttons. To get started creating pages, navigate to your Dev Portal and click **Portal Editor** in the sidebar. Pages are built using Markdown Components (MDC). Additional documentation on syntax, as well as tools for generating components, are available on a [dedicated MDC site](https://portaldocs.konghq.com/).
```

### Page structure

On the left panel inside the Portal Editor, you'll see a list of pages in your Dev Portal. The name for each page is a `slug`, and will be used to build the URL for that page. You can nest child pages under other pages. If pages are nested, the slugs will be combined to build the URL.

> For example, if the `about` has a child page, `contact`, the URL for the `contact` page would be `/about/contact`.

Reviewer: ```suggestion
For example, if `about` has a child page called `contact`, its URL will be `/about/contact`.
```

If you would like to provide a custom Open Graph image, you can specify it in the page's front matter with the `image` property as a string:

```yaml
---
title: Home
description: Start building and innovating with our APIs
image: https://example.com/images/my-image.png
---
```

If you would like more control over the image, the front matter property also accepts an object interface:

```yaml
---
title: Home
description: Start building and innovating with our APIs
image:
  url: https://example.com/images/my-image.png
  alt: A description of the image
  width: 300px
  height: 200px
---
```

## Reuse content with snippets

You can reuse content on multiple pages by using snippets. Snippets allow you to store and write content in one location and use that content in multiple pages. You can also use snippets to publish content temporarily, like system outages or special events. 

Snippets are built using Markdown Components (MDC). See the [dedicated MDC site](https://portaldocs.konghq.com/components/snippet) for more information about Snippet syntax and usage.

> To get started creating snippets, navigate to your Dev Portal and click **Portal editor**, then click **Snippets**. 

Reviewer: I'm not sure what's our established approach, but it might be a good idea to add some screenshot here.

Author: Added a screenshot in my latest commit. We typically use them sparingly, but since this is heavily UI focused, I think a screenshot could work in the overview to orient users.
